### PR TITLE
feat/ARM-278 - Added Radix checkbox input

### PR DESCRIPTION
**Pull request by tomRocketmakers**
## What's new?

- Added radix checkbox component

## Ticket number(s) in JIRA (if internal)

ARM-278

[board](https://rocketmakers.atlassian.net/jira/software/projects/HM/boards/172)

## Checklist

- [x] are your changes in Storybook?
- [x] are any breaking changes documented in `docs/migrating_from_oldstrong.md`?
- [x] does _everything_ have jsdoc?
- [x] is everything exported from index.ts?
- [x] are all new hooks added to `src/hooks/hooks.stories.mdx` or given their own docs in Storybook?
- [x] are all new SCSS mixins added to `src/theme/mixins.stories.mdx`?
- [x] are all new SCSS variables added to `src/theme/variables.stories.mdx`?

### CODE REVIEW

File: module-new/src/components/checkbox/checkbox.component.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        case 'indeterminate':
+          return globals.checkboxCustomIndeterminateIndicator;
+        default:
+          return globals.checkboxCustomIndicator;
+      }
+    }, [boundValue, globals.checkboxCustomIndicator, globals.checkboxCustomIndeterminateIndicator]);
+
+    return (
+      <StatusWrapper
+        className={concat(validationErrorsClassName, 'arm-input-base')}
```
Comment (jlews) [inline module-new/src/components/checkbox/checkbox.component.tsx:116]: Should this component receive the same class name as the validation errors? I think it needs its own class?

File: module-new/src/components/checkbox/checkbox.component.tsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+              htmlFor={id}
+            >
+              {label}
+            </Label>
+          )}
+        </div>
+        {!!bindConfig.validationErrorMessages?.length && bindConfig.shouldShowValidationErrorMessage && (
+          <ValidationErrors
+            className={validationErrorsClassName}
+            validationMode={'message'}
```
Comment (jlews) [inline module-new/src/components/checkbox/checkbox.component.tsx:154]: Hmmmm, seems a bit opinionated 😂

**Review by jlews — CHANGES_REQUESTED** [review]

File: module-new/src/components/checkbox/checkbox.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+};
+
+export const ValidationError: Story = {
+  args: {
+    label: 'Check',
+    direction: 'vertical',
+    testId: 'checkbox-container',
+  },
+  render: () => {
+    <h2>Validation mode - both</h2>;
```
Comment (jlews) [inline module-new/src/components/checkbox/checkbox.stories.tsx:71]: Not part of the return

File: module-new/src/components/checkbox/checkbox.component.tsx (hunk 1 above)
Comment (tomRocketmakers) [reply module-new/src/components/checkbox/checkbox.component.tsx:116]: Yes indeed, changed in update

File: module-new/src/components/checkbox/checkbox.component.tsx (hunk 2 above)
Comment (tomRocketmakers) [reply module-new/src/components/checkbox/checkbox.component.tsx:154]: That's just your _opinion_. (Changed in update)

File: module-new/src/components/checkbox/checkbox.stories.tsx (hunk 3 above)
Comment (tomRocketmakers) [reply module-new/src/components/checkbox/checkbox.stories.tsx:71]: Yes indeed - removed

**Review by jlews — APPROVED** [review]

**Pull request merged**